packages/backend/src/services/UserService.ts

```diff
@@ -518,7 +520,28 @@ export class UserService extends BaseService {
             return loginUser;
         }
 
-        // User already logged in? Link openid identity to logged-in user
+        let sessionUser = authenticatedUser;
+        // Link the openid identity to a user if they already have another OIDC with the same email
+        if (!authenticatedUser) {
+            const identities =
+                await this.openIdIdentityModel.findIdentitiesByEmail(
+                    openIdUser.openId.email,
+                );
+            const identitiesUsers = uniq(
+                identities.map((identity) => identity.userUuid),
+            );
+            if (identitiesUsers.length > 1) {
+                Logger.warn(
+                    `Multiple openid identities found with the same email ${openIdUser.openId.email}`,
+                );
```

Comment:
Warning to cover edge cases with multiple authentications providers enabled.

Reply:
great idea!